--- kiev020TokenManager.java.old	Thu Feb  6 19:47:28 2003
+++ kiev020TokenManager.java	Thu Feb  6 19:48:06 2003
@@ -4300,7 +4300,7 @@
 static private final Token jjFillToken()
 {
    Token t = Token.newToken(jjmatchedKind);
-   t.kind = jjmatchedKind;
+   if( t.kind == 0 ) t.kind = jjmatchedKind;
    String im = jjstrLiteralImages[jjmatchedKind];
    t.image = (im == null) ? input_stream.GetImage() : im;
    t.beginLine = input_stream.getBeginLine();
